compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -972,8 +981,22 @@ fn link_natively<'a>(
 
                 sess.abort_if_errors();
             }
-            info!("linker stderr:\n{}", escape_string(&prog.stderr));
-            info!("linker stdout:\n{}", escape_string(&prog.stdout));
+
+            if !prog.stderr.is_empty() {
+                // We already print `warning:` at the start of the diagnostic. Remove it from the linker output if present.
+                let stderr = escape_string(&prog.stderr);
+                debug!("original stderr: {stderr}");
```

Comment:
`debug!()` can't be enabled for the shipped binaries AFAIK. I think this should be kept `info!()`.

Reply:
hmm, do you expect the `warning:` stripping to be incorrect? that's the only case when the `debug` output would be different from the warning diagnostic.

Reply:
I guess it is fine to leave as is.